pkg/android/get-ndk-version.go

```diff
@@ -1,21 +1,52 @@
 package android
 
 import (
+	"bufio"
+	"fmt"
+	"os"
 	"path/filepath"
 	"strconv"
 	"strings"
 )
 
 // GetNdkVersion - Returns the major NDK version
-func GetNdkVersion(path string) (int, error) {
+func GetNdkVersion(ndkPath string) (int, error) {
+	// Construct the path to source.properties
+	sourceFile := filepath.Join(ndkPath, "source.properties")
 
-	ndkVersion := strings.Split(filepath.Base(path), ".")
+	// Open the file
+	file, err := os.Open(sourceFile)
+	if err != nil {
+		return 0, fmt.Errorf("failed to open source.properties: %v", err)
+	}
+	defer file.Close()
 
-	ndkIntVersion, err := strconv.Atoi(ndkVersion[0])
+	// Read the file line by line
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "Pkg.Revision") {
+			// Extract version string
+			parts := strings.Split(line, " = ")
```

Comment:
Technically the format doesn't require the spaces - it would be better to split on just the "=" and then trim the resulting string.

```suggestion
			parts := strings.Split(line, "=")
```